src/oscar/apps/checkout/applicator.py

```diff
@@ -0,0 +1,73 @@
+from oscar.core import prices
+
+
+class SurchargeList(list):
+    @property
+    def total(self):
+        return sum(self).price
+
+
+class SurchargeMethod(object):
+    method = None
+    price = None
+
+    def __init__(self, method, price):
+        self.method = method
+        self.price = price
+
+    def __add__(self, other):
+        price = prices.Price(
+            currency=self.price.currency or other.price.currency,
+            incl_tax=self.price.incl_tax + other.price.incl_tax,
+            excl_tax=self.price.excl_tax + other.price.excl_tax
+        )
+
+        return SurchargeMethod(self.method, price)
+
+    def __radd__(self, other):
+        if other == 0:
+            return self
+        else:
+            return self.__add__(other)
+
```

Comment:
Please move the `__add__` and `__radd__` methods to core.prices.Price

Reply:
Great suggestion